tests/utility/type.ts

```diff
@@ -90,3 +92,14 @@ test('do nothing on disabled element', async () => {
 
   expect(getEvents()).toHaveLength(0)
 })
+
+describe('on shadow DOM', () => {
+  test('type into an input element', async () => {
+    const {element, user} = setup('<shadow-input></shadow-input>')
+
+    // Skip click because delegatesFocus is not implemented in jsdom
+    await user.type(element, 'test', {skipClick: true})
```

Comment:
This only works in our specific `setup` that includes the focus workaround.
We should make sure this works inside our own tests and for our users.

Reply:
Maybe this is more related to #1002? The `click` just means it looses focus.

Reply:
Which is correct. `shadow-input` attaches the shadow DOM with `delegatesFocus: false` . Clicking the `shadow-input` outside of `input` focuses the closest focusable element - which is `document.body` here.
If this is not intended, we need to fix the component and/or our implementation instead of bending the test ;)

#1002 is environment-specific. While it still needs someone to investigate why it happens and how we might be able to fix it, it doesn't affect our tests running in Jsdom.

Reply:
I removed the test. This looked like a quick win, but it is not it seems, so removing this out of this PR.

Reply:
I added a new test for `keyboard`.